### PR TITLE
End of file as newline

### PR DESCRIPTION
**Pull request by rewinfrey**
Based on https://github.com/tree-sitter/tree-sitter-python/pull/9/files (to help verify parse errors are not related to tabs).

/cc @maxbrunsfeld

### CODE REVIEW

**Comment by maxbrunsfeld** [comment]
@rewinfrey Do we need a separate `eof_newline`, or can we just create a `newline` token at EOF?

File: src/scanner.cc [hunk 1]
```diff
@@ -73,6 +74,11 @@ struct Scanner {
       lexer->advance(lexer, true);
     }
 
+    if (lexer->lookahead == 0) {
+      lexer->result_symbol = EOF_NEWLINE;
+      return true;
+    }
```
Comment (rewinfrey) [inline src/scanner.cc]: @maxbrunsfeld I tried at first to use `NEWLINE`, but I'm not able to configure the grammar such that infinite looping doesn't occur, or that we still pass all tests and edge cases. From trying to find a way to make this work with only `NEWLINE`, I've found having a separate token indicating EOF makes this easier to reason about. Let me know what you think, or if you'd like to take a look at this together! :D
